dts/arm/renesas/ra/ra-cm4-common.dtsi

```diff
@@ -21,49 +22,51 @@
 		};
 	};
 
-	clocks {
-		mosc: mosc {
-			compatible = "fixed-clock";
+	clocks: clocks {
+		xtal: clock-xtal {
+			compatible = "renesas,ra-cgc-external-clock";
 			clock-frequency = <1200000>;
 			status = "disabled";
 			#clock-cells = <0>;
 		};
 
-		sosc: sosc {
-			compatible = "fixed-clock";
+		subclk: clock-subclk {
+			compatible = "renesas,ra-cgc-subclk";
 			clock-frequency = <32768>;
 			status = "disabled";
 			#clock-cells = <0>;
 		};
 
-		hoco: hoco {
+		hoco: clock-hoco {
 			compatible = "fixed-clock";
 			clock-frequency = <24000000>;
 			status = "okay";
 			#clock-cells = <0>;
 		};
 
-		moco: moco {
+		moco: clock-moco {
 			compatible = "fixed-clock";
 			clock-frequency = <8000000>;
 			status = "okay";
 			#clock-cells = <0>;
 		};
 
-		loco: loco {
+		loco: clock-loco {
 			compatible = "fixed-clock";
 			clock-frequency = <32768>;
 			status = "okay";
 			#clock-cells = <0>;
 		};
 
 		pll: pll {
-			compatible = "fixed-factor-clock";
-			status = "disabled";
-			clocks = <&mosc>;
-			clock-div = <2>;
-			clock-mult = <8>;
+			compatible = "renesas,ra-cgc-pll";
 			#clock-cells = <0>;
+
+			/* PLL */
+			source = <RA_PLL_SOURCE_MAIN_OSC>;
+			div = <RA_PLL_DIV_2>;
```

Comment:
wrong:

- source: define another clock and make it source using eg `clocks =  <&main_osc>;`
- divider should be given as integer `div = <2>;`.

Reply:
@duynguyenxa
Are these properties actually used?

Reply:
@soburi , these properties are used to provide information for the bsp_clock_cfg.h in the hal layer:
```
#define BSP_CFG_PLL_SOURCE                                                                         \
	BSP_CLOCK_PROP_HAS_STATUS_OKAY_OR(DT_NODELABEL(pll), source, RA_PLL_SOURCE_DISABLE)
#define BSP_CFG_PLL_DIV BSP_CLOCK_PROP_HAS_STATUS_OKAY_OR(DT_NODELABEL(pll), div, RA_PLL_DIV_2)
#if DT_NODE_HAS_STATUS(DT_NODELABEL(pll), okay)
```

The BSP_CFG_PLL_SOURCE, and BSP_CFG_PLL_DIV BSP will be used in the bsp_clock_init() to configure PLL clock.

Reply:
FYI: We can't have HALisms in DT.

Reply:
@gmarull , the macro is defined in zephyr main repo, zephyr/include/zephyr/dt-bindings/clock/ra_clock.h

Reply:
it still looks wrong, why isn't `div = <2>`? Are such values encoding what HAL expects?

Reply:
@gmarull , sorry for late response, it's expect the value map with HWM here for the pll.
![image](https://github.com/user-attachments/assets/ab410654-8351-4d46-8eba-a553568f7f3c)

Reply:
@duynguyenxa 

The root of this problem is that the DeviceTree is a hardware description language,
and how values ​​are handled should not be mixed in here.
In other words, the DeviceTree should be able to describe semantic values; for example, if the division factor is 2, then it should be describable as "2", and it should be up to the implementation side, i.e., the .h or .c file, to convert this to bit notation.

Reply:
I splitted out this issue to #78365.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/78365 is already merged.